### ISSUE TITLE
Guideposts along Hiking Routes

### ISSUE THREAD

**Issue by alpter** [waymarkedtrails/waymarked-trails-site#262]
Hi
just wondering... since many guideposts have now informations, some with code, some with pictures relating to the hiking tabs     [guide posts on OSM](http://www.weeklyosm.eu/archives/4504)
wouldnt be possible to show informations related to guide posts within the Waymarked Trails map... as a side windows similar to the Relation window -  showing info and details....

I tried to add the guideposts to relation but no information were shown...

[](http://hiking.waymarkedtrails.org/#route?id=6730399)

Thanks ...

**Comment by lonvia** [comment]
Now that waymarkedtrails has a clickable map, we might as well make the guideposts clickable and display extended information in the side panel. The main information on the guidepost node itself is fairly easy to do. The destination parts are a bit more difficult to compute.

Please don't add guideposts to route relations. It is obvious enough which guidepost belongs to a route simply because of its geographic location. Producing a list of guidepost on the route should be easily doable using the current database information.

**Comment by AlfredoSP** [comment]
In here
http://wiki.openstreetmap.org/wiki/CAI
we are recommending to add the guidepost to hike relations since to the community we reached a general consensus about it. Can you explain what kind of problem this represent?

**Comment by lonvia** [comment]
They violate the principle of "one OSM object - one real world entity". A hiking route relation should only describe the route. It's bad design to add everything remotely related to the route just because it is technically possible. It is much harder to handle for mappers (place guidepost node next to way -> easy, find all relations around guidepost and add note to relations -> difficult, remember to update relations when editing guidepost and vice versa -> really hard). And it is bound to cause trouble for data processing sooner or later. Guidepost might still look like a simple thing to add to the relation. But the next thing are all the accommodation along the route and suddenly there are (building) ways in the route that are not a walkable way and the logic to create the routes suddenly becomes complicated and involves a lot of guessing.

And as described above it is entirely unnecessary if you have a geographic database.

**Comment by AlfredoSP** [comment]
I see. Consider that at the begining I was not even thinking that the guidepost should be part of the relation, it all came after when we published the wiki
Another solution we were thinking was to use one node of the way and add to it the tags for the guidepost. Some consider this wrong as well because in practice the guidepost is on the side of way. What's you opinion on this?

**Comment by lonvia** [comment]
The most common approach these days is to put the guidepost on the side of the way where it is actually located. So, I'd probably follow that as well but essentially both works for me.

If you want to relate the routes and guide posts, have a look at http://wiki.openstreetmap.org/wiki/Relation:destination_sign which has been used for hiking guide posts as well.

**Comment by lucadelu** [comment]
@lonvia I'm sorry but I don't think your previous answer is correct, first the idea to insert the guidepost in the route relation is not violating the principle of "one OSM object - one real world entity", because a hiking route is composed by the path but also other elements, mainly guidepost, but for multi-days route also the sleeping place (not all the sleeping place along the route but the suggested one between a day and the next one) could be added, why not? 
What is the difference between a guidepost for an hiking route and a bus stop for a bus route?
However in the [hiking route place](https://wiki.openstreetmap.org/wiki/Relation:route) the guidepost is an "official" member to use in the relations.
The problem about data processing seems to me a non problem, I tried with overpass-turbo and there are no problem at all, I will do some test with osm2pgsql but I hope there will be no problem also with it (otherwise it should be a bug)

Can you explain a little bit better your last sentence "And as described above it is entirely unnecessary if you have a geographic database", because I don't find a correct geographical way to search all the guidepost assigned to only a hiking route?

**Comment by mbranco2** [comment]
@lonvia:
I agree with Luca's previous comment, and I add this observation:
I fully agree that it's good practice not to add unnecessary (redundant) information in relations (i.e. with closed ways, specify all the objects inside and outsde the perimeter, specifying the outside and inside roles for each item: it's implicit and easily derivable in the algorithm, what is inside and what is outside...)
But, for this specific issue (a route and their related guideposts)  :
rendering a route, how you can add only its related guideposts, ignoring other guideposts "very near" to the same route but not related to it?

**Comment by lonvia** [comment]
There is no need to convince me of anything. @AlfredoSP was asking me why I think it's a terrible idea and I stated my reasons. I admit that I have a rather strong opinion on the issue and it's unlikely that waymarkedtrails will ever use that information. But if you still want to add the guideposts, don't let me stop you.

**Comment by lonvia** [comment]
To add to that:

> What is the difference between a guidepost for an hiking route and a bus stop for a bus route?

The difference is that a bus route does not exist without bus stops. There simply is not point to it without the stops. (An even more puristic view would be: a bus route is only a sequence of bus stops, the ways in between are just an implementation issue and might as well be left out.) There is no problem removing all the guideposts from a hiking route. The route is still there and walkable and hasn't lost any meaning.

**Comment by mbranco2** [comment]
Sorry, I have absolutely no intention to polemicize, I'm just trying to contribute in a constructive manner to the discussion...
If we have a graph of paths in the mountains, and there are multiple ways to reach point B starting from point A, a specific route connecting A-B is just only a convention (like a bus route), and this is why is requested, to map a route, that the route itself it's permanently marked with **specific** guideposts.
In other words, hiking routes (= OSM relations with type=route and route=hiking) are virtual, the same as bus routes; physical are the single ways composing the route.

**Comment by datendelphin** [comment]
@mbranco2 The bus stop is the item I as a rider am interested in, and the time table of course. I can only get on and off the bus there, no matter what route it takes.
But it does not matter to me when hiking or cycling where and how many guide posts there are belonging to the route. I can get on and off the route any time.

**Comment by mbranco2** [comment]
Hi dantedelphin, 
the discussion was about another fact:
Sarah said "The difference is that a bus route does not exist without bus stops" , and I answered that also a hiking route does not exists without guideposts, they are virtual in the same manner.

**Comment by pelderson** [comment]
I'm not picking sides but I would like to add that many guideposts are located near totally unrelated routes.  As a hiker I see no reason for waymarkedtrails to relate specific guideposts to specificic routes, but I can think of other use cases where data users might want that information.

**Comment by alpter** [comment]
@lonvia @AlfredoSP @lucadelu @mbranco2 The reason for all this issue ... and the primary request to add it into WaymarkedTrail.... is because at least in the Piedmont Region.... but overall in other Regions..... is happening now in Ligurian Region with the AVML Trail... but in France is something now consoilidated....
the official IGM Maps in France has all the single guideposts mapped with a numbers....   
**> The guidepost are used as a "security / mountain safe / secours alpin / alpine rettung"  landmark.**
Guideposts are being seen as "safety point" with a known unique code, a known unique position and therefore strictly connected to the route you are walking.  **In case of HELP your guidepost code that is connected to your route will help you track you down.**
In particular, in Piedmont Region, single routes connect a series of guideposts, example the single route A26, has a series of guideposts, starting from the bottom point A26/1 ... up to the highest point A26/6
(https://hiking.waymarkedtrails.org/#guidepost?id=4688733660&map=18!44.1804!8.0224)

All this can be useful in case you are lost in the mountain  - or if you are calling for help -  and you can be tracked down by the nearest guidepost code .  Also can become useful in term of maintenace of guideposts or in term of planning and hike if additionally to the basic information can be added a metadata link from wikicommon with picture of the guidepost  (https://www.openstreetmap.org/node/4928595531)
Would be great if the metadata link to the wikicommons link could be seen also on WaymarkedTrails,,,
[wikimedia_commons](https://commons.wikimedia.org/wiki/File:RPE_Guide_Post_A50_-_1.jpg)

My limit of coding and technical knowledge does not allow me to understand if it ok to embed into the relation the guidepost or not.... end if  "it is bound to cause trouble for data processing sooner or later".    

In the comparison to the bus stop example, I can say that if the route and the guideposts designated are constructed together "physically", with codification of guideposts, together with the codification of the route ... then intuitively yes, they are part of the relation but only if I can preceive it on the field survey...
But for example in France, where guideposts have a numeric general number for an entier area or Region  (from 1 to 500 example) and the number is not related to the route, then embed it with the relation looks more complicated....unless that numbers are related to the hiking routes.... (we should ask IGM for this) 
See here for example :
(https://www.geoportail.gouv.fr/carte?c=7.56950097722835,44.11747516239586&z=16&l0=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN25TOUR.CV::GEOPORTAIL:OGC:WMTS(1)&permalink=yes)

**Comment by lonvia** [comment]
Hmm, we have the details view as requested. We even have the [destination part](https://hiking.waymarkedtrails.org/#guidepost?id=3904774542&map=18!46.7639!11.3742) through destination relations. Time to close this.

**Comment by alpter** [comment]
@lonvia Yep... for now look Ok. Thanks...
hey...the [guidepost you mentioned](https://www.openstreetmap.org/relation/5804993#map=18/46.76422/11.37412) is... part of a relation ;-) 
Thanks Lonvia !

**Comment by lonvia** [comment]
@alpter Yes, it's  part of a [destination_sign](https://wiki.openstreetmap.org/wiki/Relation:destination_sign) relation. They are really nice to map the information on the guide post (destinations, distance, time). And waymarkedtrails will show all that. Here is a better example with destination times: https://hiking.waymarkedtrails.org/#guidepost?id=3902994118

As some point the guide post details page will also link back to the routes for each direction. At the moment this is not possible because it gets the information from an [external side](https://osm.mueschelsoft.de/destinationsign/example/index.htm), At some point it will become part of the DB.

**Comment by alpter** [comment]
Thanks....  I looked at it and looks promising...
[OSM Destination Signs](https://osm.mueschelsoft.de/destinationsign/example/index.htm)
